libqtile/core/manager.py

```diff
@@ -189,8 +190,10 @@ def noop(qtile):
         hook.fire("startup")
 
         if state:
-            st = pickle.load(io.BytesIO(state.encode()))
             try:
+                with open(state, 'rb') as f:
+                    st = pickle.load(f)
+                os.remove(state)
```

Comment:
Can we put this in a finally: block maybe? Just to be really sure it does get deleted.